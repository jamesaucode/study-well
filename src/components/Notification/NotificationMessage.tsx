import React from "react";
import styled from "styled-components";
import { Message } from "../../styles/shared";
import { NextFC } from "next";
import { IMessage } from "../../../interfaces/message";
import SVG from "react-inlinesvg";

interface WrapperProps {
  mounted?: boolean;
}
const Wrapper = styled.div<WrapperProps>`
  display: flex;
  justify-content: center;
  border-radius: 5px;
  width: 100%;
  padding: 0.75rem;
  box-sizing: border-box;
  transition: 0.5s ease-in-out opacity;
`;
const Important = styled.span`
  font-weight: 700;
`;
const SVGWrapper = styled.div`
  padding: 0 0.5rem;
`;
const renderImportant = (text: string): React.ReactElement => (
  <Important>{text}</Important>
);
const NotificationMessage: NextFC<IMessage> = ({
  success,
  message,
  id,
  // delay = 3000,
  removeNotification,
}): JSX.Element => {
  return (
    <Wrapper key={id}>
      <Message success={success}>
        <div>
          {success ? renderImportant("Success! ") : renderImportant("Oops! ")}
          {message}
        </div>
        <SVGWrapper onClick={(): void => removeNotification(id)}>
          <SVG className="small-icon" src="/static/images/close.svg" />
        </SVGWrapper>
      </Message>
    </Wrapper>
  );
};

NotificationMessage.defaultProps = {
  success: false,
  message: "No message",
};

export default React.memo(NotificationMessage);
