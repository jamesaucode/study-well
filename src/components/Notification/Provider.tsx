import React, { useRef, useState, useEffect } from 'react';
import { Context } from './Context';

const Provider = () => {
    const root  = useRef<Node | null>(null);
    const id = useRef([]);
    const [notifications, setNotifications] = useState([]);

    useEffect(() => {
        root.current = document.createElement('div');
        document.body.appendChild(root.current);
        return () => {
           id.current.forEach(clearTimeout);
           if (root.current) {
            document.body.removeChild(root.current);
           }
        };
    }, [])

    // const remove = (notification) => {

    // }
}
