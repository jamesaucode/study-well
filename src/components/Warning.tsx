import React from "react";
import styled from 'styled-components';

interface PropTypes {
  text: string
}
const Message = styled.p`
  font-size: 0.8em;
  color: red;
`

const Warning : React.FC<PropTypes> = ({ text }) => {
  return (
    <Message>{text}</Message> 
  )
}

export default Warning;