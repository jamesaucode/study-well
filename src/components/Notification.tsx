import React, { useState, useEffect, useRef } from "react";
import NotificationMessage from "./NotificationMessage";
import { MessageType } from "../../typings/message";

const Notification = (props: any) => {
    const [notifications, setNotifications] = useState();
    const popNotification = () => {
        setNotifications(notifications.slice(0, -1));
    }
    const pushNotification = (message  = "lmaooooo", status: boolean) => {
        setNotifications(notifications.concat({
            message,
            success: status
        }))
    }

  const children = React.Children.map(props.children, child => {
      return React.cloneElement(child, {
          popNotification,
          pushNotification 
      })
  })
  if (notifications) {
    return (
      <React.Fragment>
        {notifications.map((notification: MessageType) => {
          return (
            <NotificationMessage
              message={notification.message}
              success={notification.success}
            />
          );
        })}
        {children}
      </React.Fragment>
    );
  } else {
    return <button onClick={popNotification}>Test Button</button>;
  }
};

export default React.memo(Notification);
