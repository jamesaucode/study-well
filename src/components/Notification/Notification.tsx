import React, { useState, useEffect, useRef, useMemo } from "react";
import NotificationMessage from "./NotificationMessage";
import { MessageType } from "../../../typings/message";
import styled from "styled-components";

const uuid = require("uuid");

const AllNotificationsWrapper = styled.div`
  height: 0; // Hacky ?
  top: 20px;
  position: sticky;
  z-index:10;
`;
const Notification = (props: any) => {
  const [notifications, setNotifications] = useState<MessageType[]>([]);
  const delay = 5000;
  const id = useRef<number[]>([]);
  useEffect(() => {
    return () => {
      id.current.forEach(clearTimeout);
    };
  }, []);
  const removeNotification = (notificationId: string) => {
    setNotifications(prevState => {
      console.log("Removing Notification");
      return prevState.filter(n => n.id !== notificationId);
    });
  };
  const pushNotification = (message = "Lol", status: boolean) => {
    const notification = {
      message,
      success: status,
      id: uuid(),
      delay,
    };
    console.log("Setting up the timer");
    const timerId = setTimeout(() => {
      console.log("Now removing the notification");
      removeNotification(notification.id);
      id.current.splice(id.current.indexOf(timerId), 1);
    }, delay);
    id.current.push(timerId);
    setNotifications(allNotification => allNotification.concat(notification));
    return notification;
  };

  const children = React.Children.map(props.children, child => {
    return React.cloneElement(child, {
      pushNotification,
      removeNotification,
    });
  });
  if (notifications) {
    return (
      <React.Fragment>
        <AllNotificationsWrapper>
          {notifications.map((notification: MessageType) => {
            return (
              <NotificationMessage
                message={notification.message}
                success={notification.success}
                id={notification.id}
                delay={notification.delay}
                removeNotification={removeNotification}
              />
            );
          })}
        </AllNotificationsWrapper>
        {children}
      </React.Fragment>
    );
  } else {
    return <React.Fragment>{children}</React.Fragment>;
  }
};

export default React.memo(Notification);
