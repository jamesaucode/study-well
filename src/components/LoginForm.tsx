import React, { useState, useEffect, useRef } from "react";
import { NextFC } from "next";
import styled from "styled-components";
import fetch from "isomorphic-unfetch";
import Register from "./Register";
import { InputValidator } from "../../services/validation.service";
import { FormBottom } from "../styles/shared";
import Warning from "./Warning";

const googleLoginButton = require("../../static/images/btn_google_signin_dark_normal_web@2x.png");
const maxFormWidth = "400px";

const LoginButton = styled.a`
  max-width: 200px;
  cursor: pointer;
  margin: 1rem;
`;

const StyledLink = styled.a`
  color: blue;
  text-decoration: underline;
  cursor: pointer;
  font-weight: bold;
`;

const LoginButtonLogo = styled.img`
  width: 100%;
  height: 100%;
`;
const Divider = styled.div`
  width: 100%;
  max-width: ${maxFormWidth};
  height: 0.6em;
  border-bottom: 1px solid #dddddd;
  text-align: center;
  margin-bottom: 2rem;
`;
const DividerText = styled.span`
  font-size: 0.7em;
  font-weight: 600;
  background: white;
  padding: 0 1rem;
  color: ${({ theme }) => theme.mainfc};
`;
interface FormInputProps {
  validated: boolean;
}
const FormInput = styled.input<FormInputProps>`
  border: 1px solid #aaaaaa;
  border-radius: 2px;
  padding: 0.75rem;
  margin: 0.5rem 0;
  width: 100%;
  font-size: 0.7em;
  box-sizing: border-box;
  &:focus {
    border: ${({ validated }) =>
      validated ? "1px solid #8610f9" : "1px solid red"};
  }
`;
const FormWrapper = styled.div`
  max-width: ${maxFormWidth};
  text-align: center;
`;
const FormSubmit = styled.button`
  color: white;
  font-size: 0.7em;
  font-weight: 600;
  width: 100%;
  border: none;
  border-radius: 3px;
  padding: 0.65rem;
  background: #8610f9;
  opacity: ${({ disabled }) => (disabled ? 0.3 : 1)};
  cursor: ${({ disabled }) => (disabled ? "not-allowed" : "pointer")};
`;
const Wrapper = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
  height: fit-content;
  max-width: 400px;
  font-size: calc(0.35vw + 16px);
  padding: 1em;
`;

const Login: NextFC = () => {
  const [emailInput, setUsernameInput] = useState("");
  const [passwordInput, setPasswordInput] = useState("");
  const [showSignIn, setShowSignIn] = useState(true);
  const [showWarning, setShowWarning] = useState(false);
  const emailInputRef = useRef<HTMLInputElement>();
  useEffect(() => {
    // Focuses on the emailinputref as the component mount
    if (emailInputRef.current) {
      emailInputRef.current.focus();
    }
    return () => {
    };
  }, [])
  const handleSubmit = () => {
    fetch("/api/login", {
      method: "POST",
      credentials: "include",
      headers: {
        "Content-Type": "application/json"
      },
      body: JSON.stringify({
        email: emailInput,
        password: passwordInput
      })
    }).then(response => {
      if (response.ok && response.redirected) {
        window.location.href = "/user/cards";
      } else {
        setShowWarning(true);
        setPasswordInput("");
      }
    });
  };
  const validateEmail = () => InputValidator.email(emailInput);
  const validatePassword = () => InputValidator.password(passwordInput);
  const validateInput = () => {
    return validateEmail() && validatePassword();
  };
  const handleKeyDown = (event : React.KeyboardEvent) => {
    const isValid = validateInput();
    if (event.keyCode === 13 && isValid) {
      handleSubmit();
    }
  }
  const handleClick = () => {
    setShowSignIn(!showSignIn);
  };

  return showSignIn ? (
    <Wrapper>
      <LoginButton href="/auth/google">
        <LoginButtonLogo src={googleLoginButton} />
      </LoginButton>
      <Divider>
        <DividerText>or</DividerText>
      </Divider>
      <FormWrapper>
        {showWarning && <Warning text={"Incorrect Credentials"} />}
        <FormInput
          onChange={e => {
            setUsernameInput(e.target.value);
          }}
          ref={emailInputRef}
          onKeyDown={handleKeyDown}
          validated={validateEmail()}
          value={emailInput}
          placeholder="Username / Email"
          name="username"
          type="email"
        />
        <FormInput
          onChange={e => {
            setPasswordInput(e.target.value);
          }}
          onKeyDown={handleKeyDown}
          value={passwordInput}
          validated={validatePassword()}
          placeholder="Password"
          name="password"
          type="password"
        />
        <FormSubmit disabled={!validateInput()} onClick={handleSubmit}>
          Login
        </FormSubmit>
        <FormBottom>
          New user ? <StyledLink id="signup" onClick={handleClick}>Signup </StyledLink>
          here!
        </FormBottom>
      </FormWrapper>
    </Wrapper>
  ) : (
    <Register
      toggleForm={() => {
        setShowSignIn(true);
      }}
    />
  );
};

export default Login;
