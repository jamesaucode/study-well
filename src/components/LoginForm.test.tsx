/* eslint-disable */
import LoginForm from "./LoginForm";
import Warning from "./Warning";
import { NotificationProvider } from "./Notification/Notification";
import { mount } from "enzyme";

describe("LoginForm Tests", () => {
  it("Component Renders", () => {
    const wrapper = mount(
      <NotificationProvider>
        <LoginForm />
      </NotificationProvider>,
    );
    expect(wrapper).toMatchSnapshot();
  });

  it("Changehandler of the component works correctly", () => {
    const wrapper = mount(
      <NotificationProvider>
        <LoginForm />
      </NotificationProvider>,
    );
    wrapper.find("input[name='username']").simulate("change", {
      target: { value: "Hello world", name: "username" },
    });
    expect(wrapper.find("input[name='username']").prop("value")).toBe(
      "Hello world",
    );
    wrapper.find("input[name='password']").simulate("change", {
      target: { value: "This is a test", name: "password" },
    });
    expect(wrapper.find("input[name='password']").prop("value")).toBe(
      "This is a test",
    );
  });

  it("Login button should be disabled if input is not validated, and vice versa", () => {
    const wrapper = mount(
      <NotificationProvider>
        <LoginForm />
      </NotificationProvider>,
    );
    expect(wrapper.find("button").prop("disabled")).toBeTruthy();
    wrapper.find("input[name='username']").simulate("change", {
      target: { value: "testing@gmail.com", name: "username" },
    });
    wrapper.find("input[name='password']").simulate("change", {
      target: { value: "testingpassword", name: "password" },
    });
    expect(wrapper.find("button").prop("disabled")).toBeFalsy();
  });

  it("Warning should not display on mount, and should display after first failed login attempt", () => {
    const wrapper = mount(
      <NotificationProvider>
        <LoginForm />
      </NotificationProvider>,
    );
    expect(wrapper.find(Warning).exists()).toBeFalsy();
    wrapper.find("input[name='username']").simulate("change", {
      target: { value: "testing@gmail.com", name: "username" },
    });
    wrapper.find("input[name='password']").simulate("change", {
      target: { value: "testingpassword", name: "password" },
    });
    /** Need to figure out how to make the api call */
  });

  it("Should show login form on mount, show signup form if user clicks the signup link", () => {
    const wrapper = mount(
      <NotificationProvider>
        <LoginForm />
      </NotificationProvider>,
    );
    expect(wrapper.find("style__FormBottom").text()).toBe(
      "New user ? Signup here!",
    );
    wrapper.find("LoginForm__StyledLink").simulate("click");
    expect(wrapper.find("style__FormBottom").text()).toBe(
      "New user ? Login here!",
    );
  });
});
