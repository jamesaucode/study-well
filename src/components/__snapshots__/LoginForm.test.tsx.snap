// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginForm Tests Component Renders 1`] = `
<Login>
  <LoginForm__Wrapper>
    <StyledComponent
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "LoginForm__Wrapper-zzt754-8",
            "isStatic": true,
            "lastClassName": "kgJUNV",
            "rules": Array [
              "display:flex;flex-direction:column;align-items:center;height:fit-content;max-width:400px;font-size:calc(0.35vw + 16px);padding:1em;",
            ],
          },
          "displayName": "LoginForm__Wrapper",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "LoginForm__Wrapper-zzt754-8",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
    >
      <div
        className="LoginForm__Wrapper-zzt754-8 kgJUNV"
      >
        <LoginForm__LoginButton
          href="/auth/google"
        >
          <StyledComponent
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "LoginForm__LoginButton-zzt754-0",
                  "isStatic": true,
                  "lastClassName": "btamXy",
                  "rules": Array [
                    "max-width:200px;cursor:pointer;margin:1rem;",
                  ],
                },
                "displayName": "LoginForm__LoginButton",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "LoginForm__LoginButton-zzt754-0",
                "target": "a",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
            href="/auth/google"
          >
            <a
              className="LoginForm__LoginButton-zzt754-0 btamXy"
              href="/auth/google"
            >
              <LoginForm__LoginButtonLogo
                src={
                  Object {
                    "process": [Function],
                  }
                }
              >
                <StyledComponent
                  forwardedComponent={
                    Object {
                      "$$typeof": Symbol(react.forward_ref),
                      "attrs": Array [],
                      "componentStyle": ComponentStyle {
                        "componentId": "LoginForm__LoginButtonLogo-zzt754-2",
                        "isStatic": true,
                        "lastClassName": "hIWQXf",
                        "rules": Array [
                          "width:100%;height:100%;",
                        ],
                      },
                      "displayName": "LoginForm__LoginButtonLogo",
                      "foldedComponentIds": Array [],
                      "render": [Function],
                      "styledComponentId": "LoginForm__LoginButtonLogo-zzt754-2",
                      "target": "img",
                      "toString": [Function],
                      "warnTooManyClasses": [Function],
                      "withComponent": [Function],
                    }
                  }
                  forwardedRef={null}
                  src={
                    Object {
                      "process": [Function],
                    }
                  }
                >
                  <img
                    className="LoginForm__LoginButtonLogo-zzt754-2 hIWQXf"
                    src={
                      Object {
                        "process": [Function],
                      }
                    }
                  />
                </StyledComponent>
              </LoginForm__LoginButtonLogo>
            </a>
          </StyledComponent>
        </LoginForm__LoginButton>
        <LoginForm__Divider>
          <StyledComponent
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "LoginForm__Divider-zzt754-3",
                  "isStatic": true,
                  "lastClassName": "iwtJTG",
                  "rules": Array [
                    "width:100%;max-width:",
                    "400px",
                    ";height:0.6em;border-bottom:1px solid #dddddd;text-align:center;margin-bottom:2rem;",
                  ],
                },
                "displayName": "LoginForm__Divider",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "LoginForm__Divider-zzt754-3",
                "target": "div",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
          >
            <div
              className="LoginForm__Divider-zzt754-3 iwtJTG"
            >
              <LoginForm__DividerText>
                <StyledComponent
                  forwardedComponent={
                    Object {
                      "$$typeof": Symbol(react.forward_ref),
                      "attrs": Array [],
                      "componentStyle": ComponentStyle {
                        "componentId": "LoginForm__DividerText-zzt754-4",
                        "isStatic": false,
                        "lastClassName": "heqDou",
                        "rules": Array [
                          "font-size:0.7em;font-weight:600;background:white;padding:0 1rem;color:",
                          [Function],
                          ";",
                        ],
                      },
                      "displayName": "LoginForm__DividerText",
                      "foldedComponentIds": Array [],
                      "render": [Function],
                      "styledComponentId": "LoginForm__DividerText-zzt754-4",
                      "target": "span",
                      "toString": [Function],
                      "warnTooManyClasses": [Function],
                      "withComponent": [Function],
                    }
                  }
                  forwardedRef={null}
                >
                  <span
                    className="LoginForm__DividerText-zzt754-4 heqDou"
                  >
                    or
                  </span>
                </StyledComponent>
              </LoginForm__DividerText>
            </div>
          </StyledComponent>
        </LoginForm__Divider>
        <LoginForm__FormWrapper>
          <StyledComponent
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "LoginForm__FormWrapper-zzt754-6",
                  "isStatic": true,
                  "lastClassName": "cJOUov",
                  "rules": Array [
                    "max-width:",
                    "400px",
                    ";text-align:center;",
                  ],
                },
                "displayName": "LoginForm__FormWrapper",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "LoginForm__FormWrapper-zzt754-6",
                "target": "div",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
          >
            <div
              className="LoginForm__FormWrapper-zzt754-6 cJOUov"
            >
              <LoginForm__FormInput
                name="username"
                onChange={[Function]}
                onKeyDown={[Function]}
                placeholder="Username / Email"
                type="email"
                validated={false}
                value=""
              >
                <StyledComponent
                  forwardedComponent={
                    Object {
                      "$$typeof": Symbol(react.forward_ref),
                      "attrs": Array [],
                      "componentStyle": ComponentStyle {
                        "componentId": "LoginForm__FormInput-zzt754-5",
                        "isStatic": false,
                        "lastClassName": "lkxRMC",
                        "rules": Array [
                          "border:1px solid #aaaaaa;border-radius:2px;padding:0.75rem;margin:0.5rem 0;width:100%;font-size:0.7em;box-sizing:border-box;&:focus{border:",
                          [Function],
                          ";}",
                        ],
                      },
                      "displayName": "LoginForm__FormInput",
                      "foldedComponentIds": Array [],
                      "render": [Function],
                      "styledComponentId": "LoginForm__FormInput-zzt754-5",
                      "target": "input",
                      "toString": [Function],
                      "warnTooManyClasses": [Function],
                      "withComponent": [Function],
                    }
                  }
                  forwardedRef={
                    Object {
                      "current": <input
                        class="LoginForm__FormInput-zzt754-5 lkxRMC"
                        name="username"
                        placeholder="Username / Email"
                        type="email"
                        value=""
                      />,
                    }
                  }
                  name="username"
                  onChange={[Function]}
                  onKeyDown={[Function]}
                  placeholder="Username / Email"
                  type="email"
                  validated={false}
                  value=""
                >
                  <input
                    className="LoginForm__FormInput-zzt754-5 lkxRMC"
                    name="username"
                    onChange={[Function]}
                    onKeyDown={[Function]}
                    placeholder="Username / Email"
                    type="email"
                    value=""
                  />
                </StyledComponent>
              </LoginForm__FormInput>
              <LoginForm__FormInput
                name="password"
                onChange={[Function]}
                onKeyDown={[Function]}
                placeholder="Password"
                type="password"
                validated={false}
                value=""
              >
                <StyledComponent
                  forwardedComponent={
                    Object {
                      "$$typeof": Symbol(react.forward_ref),
                      "attrs": Array [],
                      "componentStyle": ComponentStyle {
                        "componentId": "LoginForm__FormInput-zzt754-5",
                        "isStatic": false,
                        "lastClassName": "lkxRMC",
                        "rules": Array [
                          "border:1px solid #aaaaaa;border-radius:2px;padding:0.75rem;margin:0.5rem 0;width:100%;font-size:0.7em;box-sizing:border-box;&:focus{border:",
                          [Function],
                          ";}",
                        ],
                      },
                      "displayName": "LoginForm__FormInput",
                      "foldedComponentIds": Array [],
                      "render": [Function],
                      "styledComponentId": "LoginForm__FormInput-zzt754-5",
                      "target": "input",
                      "toString": [Function],
                      "warnTooManyClasses": [Function],
                      "withComponent": [Function],
                    }
                  }
                  forwardedRef={null}
                  name="password"
                  onChange={[Function]}
                  onKeyDown={[Function]}
                  placeholder="Password"
                  type="password"
                  validated={false}
                  value=""
                >
                  <input
                    className="LoginForm__FormInput-zzt754-5 lkxRMC"
                    name="password"
                    onChange={[Function]}
                    onKeyDown={[Function]}
                    placeholder="Password"
                    type="password"
                    value=""
                  />
                </StyledComponent>
              </LoginForm__FormInput>
              <LoginForm__FormSubmit
                disabled={true}
                onClick={[Function]}
              >
                <StyledComponent
                  disabled={true}
                  forwardedComponent={
                    Object {
                      "$$typeof": Symbol(react.forward_ref),
                      "attrs": Array [],
                      "componentStyle": ComponentStyle {
                        "componentId": "LoginForm__FormSubmit-zzt754-7",
                        "isStatic": false,
                        "lastClassName": "czvsHN",
                        "rules": Array [
                          "color:white;font-size:0.7em;font-weight:600;width:100%;border:none;border-radius:3px;padding:0.65rem;background:#8610f9;opacity:",
                          [Function],
                          ";cursor:",
                          [Function],
                          ";",
                        ],
                      },
                      "displayName": "LoginForm__FormSubmit",
                      "foldedComponentIds": Array [],
                      "render": [Function],
                      "styledComponentId": "LoginForm__FormSubmit-zzt754-7",
                      "target": "button",
                      "toString": [Function],
                      "warnTooManyClasses": [Function],
                      "withComponent": [Function],
                    }
                  }
                  forwardedRef={null}
                  onClick={[Function]}
                >
                  <button
                    className="LoginForm__FormSubmit-zzt754-7 czvsHN"
                    disabled={true}
                    onClick={[Function]}
                  >
                    Login
                  </button>
                </StyledComponent>
              </LoginForm__FormSubmit>
              <shared__FormBottom>
                <StyledComponent
                  forwardedComponent={
                    Object {
                      "$$typeof": Symbol(react.forward_ref),
                      "attrs": Array [],
                      "componentStyle": ComponentStyle {
                        "componentId": "shared__FormBottom-sc-1yigvbr-1",
                        "isStatic": false,
                        "lastClassName": "bdCQDx",
                        "rules": Array [
                          "font-size:0.8em;padding:1rem 0;color:",
                          [Function],
                          ";",
                        ],
                      },
                      "displayName": "shared__FormBottom",
                      "foldedComponentIds": Array [],
                      "render": [Function],
                      "styledComponentId": "shared__FormBottom-sc-1yigvbr-1",
                      "target": "p",
                      "toString": [Function],
                      "warnTooManyClasses": [Function],
                      "withComponent": [Function],
                    }
                  }
                  forwardedRef={null}
                >
                  <p
                    className="shared__FormBottom-sc-1yigvbr-1 bdCQDx"
                  >
                    New user ? 
                    <LoginForm__StyledLink
                      id="signup"
                      onClick={[Function]}
                    >
                      <StyledComponent
                        forwardedComponent={
                          Object {
                            "$$typeof": Symbol(react.forward_ref),
                            "attrs": Array [],
                            "componentStyle": ComponentStyle {
                              "componentId": "LoginForm__StyledLink-zzt754-1",
                              "isStatic": true,
                              "lastClassName": "jfVguh",
                              "rules": Array [
                                "color:blue;text-decoration:underline;cursor:pointer;font-weight:bold;",
                              ],
                            },
                            "displayName": "LoginForm__StyledLink",
                            "foldedComponentIds": Array [],
                            "render": [Function],
                            "styledComponentId": "LoginForm__StyledLink-zzt754-1",
                            "target": "a",
                            "toString": [Function],
                            "warnTooManyClasses": [Function],
                            "withComponent": [Function],
                          }
                        }
                        forwardedRef={null}
                        id="signup"
                        onClick={[Function]}
                      >
                        <a
                          className="LoginForm__StyledLink-zzt754-1 jfVguh"
                          id="signup"
                          onClick={[Function]}
                        >
                          Signup 
                        </a>
                      </StyledComponent>
                    </LoginForm__StyledLink>
                    here!
                  </p>
                </StyledComponent>
              </shared__FormBottom>
            </div>
          </StyledComponent>
        </LoginForm__FormWrapper>
      </div>
    </StyledComponent>
  </LoginForm__Wrapper>
</Login>
`;
